File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/WritePropertyListValuesRequest.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
import org.opcfoundation.webserver.types.digitaltwin.ObjectServiceContext;
import org.opcfoundation.webserver.types.common.UaChildId;

import java.util.Map;

public class WritePropertyListValuesRequest {
    private final String id;
    private final Map<String, Variant> propertyIdsAndValues;
    private final Map<UaChildId, Variant> subPropertyIdsAndValues;
    private final ObjectServiceContext context;

    public WritePropertyListValuesRequest(
            ObjectServiceContext context,
            Map<String, Variant> propertyIdsAndValues,
            Map<UaChildId, Variant> subPropertyIdsAndValues)
    {
        this.id = context.getObjectId().getId();
        this.propertyIdsAndValues = propertyIdsAndValues;
        this.subPropertyIdsAndValues = subPropertyIdsAndValues;
        this.context = context;
    }

    public String getId() {
        return id;
    }

    public Map<String, Variant> getPropertyIdsAndValues() {
        return propertyIdsAndValues;
    }

    public Map<UaChildId, Variant> getSubPropertyIdsAndValues() {
        return subPropertyIdsAndValues;
    }

    public ObjectServiceContext getContext() {
        return context;
    }
}

File: opcua-webservice-node/src/server/service/message/digitialtwin/WritePropertyListValuesRequest.ts
import { UaVariant } from "opcua-webapi-ts";
import { UaChildId } from "../../../types/common/UaChildId";
import { ObjectServiceContext } from "../../../types/digitaltwin/ObjectServiceContext";

export class WritePropertyListValuesRequest {
    private readonly _id: string;
    private readonly _propertyIdsAndValues: Map<string, UaVariant>;
    private readonly _subPropertyIdsAndValues: Map<UaChildId, UaVariant>;
    private readonly _context: ObjectServiceContext;

    constructor(
        context: ObjectServiceContext,
        propertyIdsAndValues: Map<string, UaVariant>,
        subPropertyIdsAndValues: Map<UaChildId, UaVariant>) {
        this._id = context.objectId.id;
        this._propertyIdsAndValues = propertyIdsAndValues;
        this._subPropertyIdsAndValues = subPropertyIdsAndValues;
        this._context = context;
    }

    get id(): string {
        return this._id;
    }

    get propertyIdsAndValues(): Map<string, UaVariant> {
        return this._propertyIdsAndValues;
    }

    get subPropertyIdsAndValues(): Map<UaChildId, UaVariant> {
        return this._subPropertyIdsAndValues;
    }

    get context(): ObjectServiceContext {
        return this._context;
    }
}